src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { NewCommentForm } from '../NewCommentForm';
+import { Post } from '../../types/Post';
+import { deleteComment, getPostComments } from '../../api/posts';
+import cm from '../../types/Comment';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<cm[]>([]);
+  const [commentsLoadingError, setCommentsLoadingError] = useState(false);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [isCommentSectionShow, setIsCommentSectionShow] = useState(false);
+  const [commentsLoaded, setCommentLoaded] = useState(false);
+  const [isDeletionError, setIsDeletionError] = useState(false);
+  const isCommentsButtonShow = !isCommentSectionShow
+    && !commentsLoadingError
+    && !isCommentsLoading;
+  const isCommentsLoaded = commentsLoaded && !commentsLoadingError;
+  const isLoadingSuccessful = isCommentsLoading && !commentsLoadingError;
+
+  const getCommentsFromServer = async () => {
+    setComments([]);
+    try {
+      if (post) {
+        setCommentLoaded(false);
+        setIsCommentsLoading(true);
+
+        const commentsFromServer = await getPostComments(post.id);
+
+        setComments(commentsFromServer);
+        setIsCommentsLoading(false);
+        setCommentLoaded(true);
+      }
+    } catch {
+      setCommentsLoadingError(true);
+    }
+  };
+
+  useEffect(() => {
+    getCommentsFromServer();
+  }, [post]);
+
+  useEffect(() => {
+    if (isCommentSectionShow) {
+      setIsCommentSectionShow(false);
+    }
+  }, [post?.id]);
+
+  const handleDeleteComment = async (commentId: number) => {
+    const commentToDelete = comments.find(comment => comment.id === commentId);
+
+    setComments(prevComments => prevComments
+      .filter(comment => comment.id !== commentId));
+    setIsCommentSectionShow(false);
+    try {
+      await deleteComment(commentId);
```

Comment:
after deleting comment you still have to show form(not button 'Write a comment')